lib/IRGen/GenValueWitness.cpp

```diff
@@ -1031,6 +1041,9 @@ getAddrOfKnownValueWitnessTable(IRGenModule &IGM, CanType type,
     if (auto enumDecl = dyn_cast<EnumDecl>(nom))
       if (!enumDecl->isObjC() && !type->isUninhabited())
         return {};
+    if (auto structDecl = dyn_cast<StructDecl>(nom))
+      if (structDecl->getClangDecl())
+        return {};
```

Comment:
I don't understand this change. Why is the logic below incorrect? It implements important optimizations for code size.

Reply:
To be honest, I'm not sure why I made this change. Everything seems to work fine without the early-exit and I can't think of a reason that it would be needed. 

I don't think there's a behavioral change here for address-only types (it is odd that there were no test failures for loadable types, though). Address only types (as addressed above) aren't POD types and also aren't empty or refcounted. So they're not going to take any of these paths.

Reply:
Ah, wait, I think I know why I had this. Because if address only types are treated as POD types, then of course this optimization is going to break things. But obviously that is now moot if we make address only types _not_-POD.